src/Core/Utility.js

```diff
@@ -0,0 +1,164 @@
+/**
+ * @constructor
+ */
+function List () {
+	this.next = this
+	this.prev = this
+	this.length = 0
+}
+List.prototype = {
+	/**
+	 * @param {Element} element
+	 * @param {Element} sibling
+	 * @return {Element}
+	 */
+	insert: function insert (element, sibling) {
+		element.next = sibling
+		element.prev = sibling.prev
+		sibling.prev.next = element
+		sibling.prev = element
+		this.length++
+		
+		return element
+	},
+	/**
+	 * @param {Element} element
+	 * @return {Element}
+	 */
+	remove: function remove (element) {
+		if (this.length < 1) 
+			return
```

Comment:
not throw?

Reply:
Follows the semantics of a List data-structure. If the list is empty exit early and don't throw.